test/TestInternalDtosRefOrleans/ClassReferencingOrleansTypeDto.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using Orleans;
+
+namespace UnitTests.DtosRefOrleans
+{
+    [Serializable]
+    public class ClassReferencingOrleansTypeDto
+    {
+        static ClassReferencingOrleansTypeDto()
```

Comment:
It looks like this static constructor is breaking the test. If I remove it or make it non-static, the test passes fine.

I'm not convinced this is an Orleans issue. Static constructors with cross-assembly dependencies are dangerous due to non-deterministic order of their execution.